icu4c/source/common/locdispnames.cpp

```diff
@@ -506,6 +507,20 @@ uloc_getDisplayName(const char *locale,
         return 0;
     }
 
+    // For the display name, we treat this as unknown language (ICU-20273).
+    static const char UND[] = "und";
+    CharString und;
+    if (locale == NULL || *locale == '\0') {
```

Comment:
Changing NULL to "und" might go too far. Please check how it behaved before.
I would expect NULL to become the default locale as usual.
There might not be a test case for it because the display name would depend on the system's default locale...